Artifacts/windows-7zip/ChocolateyPackageInstaller.ps1

```diff
@@ -0,0 +1,262 @@
+##################################################################################################
+
+# 
+# Description:
+#  - Displays the script argument values (default or user-supplied).
+#
+# Parameters:
+#  - N/A.
+#
+# Return:
+#  - N/A.
+#
+# Notes:
+#  - Please ensure that the Initialize() method has been called at least once before this 
+#    method. Else this method can only write to console and not to log files. 
+#
+
+function DisplayArgValues
+{
+    WriteLog "========== Configuration =========="
+    WriteLog $("RawPackagesList : " + $RawPackagesList)
+    WriteLog "========== Configuration =========="
+}
+
+##################################################################################################
+
+# 
+# Description:
+#  - Creates the folder structure which'll be used for dumping logs generated by this script and
+#    the logon task.
+#
+# Parameters:
+#  - N/A.
+#
+# Return:
+#  - N/A.
+#
+# Notes:
+#  - N/A.
+#
+
+function InitializeFolders
+{
+    if ($false -eq (Test-Path -Path $ChocolateyPackageInstallerFolder))
+    {
+        New-Item -Path $ChocolateyPackageInstallerFolder -ItemType directory | Out-Null
+    }
+}
+
+##################################################################################################
+
+# 
+# Description:
+#  - Writes specified string to the console as well as to the script log (indicated by $ScriptLog).
+#
+# Parameters:
+#  - $message: The string to write.
+#
+# Return:
+#  - N/A.
+#
+# Notes:
+#  - N/A.
+#
+
+function WriteLog
+{
+    Param(
+        <# Can be null or empty #> $message
+    )
+
+    $timestampedMessage = $("[" + [System.DateTime]::Now + "] " + $message) | % {  
+        Write-Host -Object $_
+        Out-File -InputObject $_ -FilePath $ScriptLog -Append
+    }
+}
+
+##################################################################################################
+
+# 
+# Description:
+#  - Installs the chocolatey package manager.
+#
+# Parameters:
+#  - N/A.
+#
+# Return:
+#  - If installation is successful, then nothing is returned.
+#  - Else a detailed terminating error is thrown.
+#
+# Notes:
+#  - @TODO: Write to $chocolateyInstallLog log file.
+#  - @TODO: Currently no errors are being written to the log file ($chocolateyInstallLog). This needs to be fixed.
+#
+
+function InstallChocolatey
+{
+    Param(
+        [ValidateNotNullOrEmpty()] $chocolateyInstallLog
+    )
+
+    WriteLog "Installing Chocolatey..."
+
+    Invoke-Expression ((new-object net.webclient).DownloadString('https://chocolatey.org/install.ps1')) | Out-Null
+
+    WriteLog "Success."
+}
+
+##################################################################################################
+
+#
+# Description:
+#  - Installs the specified chocolatet packages on the machine.
+#
+# Parameters:
+#  - N/A.
+#
+# Return:
+#  - N/A.
+#
+# Notes:
+#  - N/A.
+#
+
+function InstallPackages
+{
+    Param(
+        [ValidateNotNullOrEmpty()][string] $packagesList
+    )
+
+    $Separator = @(";",",")
```

Comment:
IIRC we include this file in each artifact that uses Chocolatey.   Should we find a way to source it from a common location instead of including it in each artifact?  That would reduce any effort needed when this script changes.